Metadata with Auth endpoint

danielfdsilva opened the pull request:
With the recent merges into develop and all the branching that was done, it became impossible to create a straightforward merge, so I extracted the relevant commits into a new branch directly from develop.

Contains the code from:
https://github.com/openaq/openaq-api/pull/397
https://github.com/openaq/openaq-api/pull/399

Still todo:
- [x] Merge locations and metadata list endpoint as suggested in https://github.com/openaq/openaq-api/pull/399#issuecomment-502791449

danielfdsilva commented:
@sethvincent This is good to go.
It contains the auth0 setup and the metadata endpoint.
The `metatada/list` was removed in favor of a `?metadata=true` flag on the locations. When using this a property `metadata` is attached to each location. If there's no metadata the value will be `null`

sethvincent commented on api/routes/locations-metadata.js:
```diff
@@ -0,0 +1,97 @@
+    config: {
+      description: 'Metadata associated with a given location'
+    },
+    handler: async function (request, reply) {
```
Ideally this endpoint would also include the `locations` record for this `id`. This is useful when visiting the metadata view/edit pages directly and need the contextual location information.

danielfdsilva replied:
implemented

danielfdsilva commented:
Note: Rebased to get the latest from `develop`

- Implemented `siteType` and `activationDate` filters.

Some questions regarding the other ones:
**inletHeight**
The `inletHeight` is a property of the `instrument` meaning that there can be several. When querying for this, should we take all the values into account and return if one matches?

**metadata completeness**
It seems to be possible to get all the keys of a particular object, but we'd need to add a function to postgres, and I think it would probably slow down the queries quite a lot...
It is easy to check for locations that have no metadata, but checking for completeness doesn't seem trivial. Would completeness mean all the properties on all the instruments? We're talking about json schema validation via sql, which would probably lead to performance problems.
Maybe we could compute this property when saving a given location and store it? If well thought this could also be the solution for having % of completeness.

Thoughts?
cc @sethvincent

kamicut commented:
Not sure if this is correct but should the new filters like activationDate and inletHeight be part of the JSDocString?

kamicut commented:
@danielfdsilva something that I was discussing with @sethvincent is what to do with metadata that has NULLs. Right now they would not show up in the API with some default parameters for searching (i.e the full range of possibilities). Should NULL data be included in the response?

danielfdsilva commented:
@kamicut we talked about this during the scrum, and concluded that the best course of action is to consider the Nulls as zeros.
I'll be submitting a fix for that.

cc @sethvincent

jflasher commented:
Thanks, I will try to check it out towards the end of next week. I'm currently on vacation and then traveling for work for a bit.

sethvincent commented:
@jflasher thanks, I made a pr for openaq-fetch: https://github.com/openaq/openaq-fetch/pull/621

sethvincent commented:
@jflasher anything else we need for this to go onto staging?

jflasher commented:
I don’t think so!


On July 19, 2019 at 15:02:32, Seth Vincent (notifications@github.com(mailto:notifications@github.com)) wrote:

>  
> @jflasher(https://github.com/jflasher) anything else we need for this to go onto staging?
>  
>  
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub(https://github.com/openaq/openaq-api/pull/403?email_source=notifications&email_token=AAGPIJQQQX2JZWBN6S3LCY3QAI2XRA5CNFSM4HZL54X2YY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGOD2M4FOY#issuecomment-513393339), or mute the thread(https://github.com/notifications/unsubscribe-auth/AAGPIJS6ATOJL6G4ATHVHODQAI2XRANCNFSM4HZL54XQ).
>  
>  
>

jflasher commented:
@sethvincent sorry if the other message wasn't clear. I think you're good to merge whenever you want. I am on a plane and getting this message out is about the limit of my bandwidth. ✈️
